test/CMakeLists.txt

```diff
@@ -0,0 +1,20 @@
+cmake_minimum_required(VERSION 3.22)
```

Comment:
I suggest placing the tests inside the "source" folder. This will make it easier to review tests when adding a new feature, along with its corresponding test folder. There will be no need to search in a separate folder. Additionally, unit tests are generally considered part of the source code.

- source
  - feature1
    - tests
  - feature2
    - tests